src/core/frames/frame_controller.ts

```diff
@@ -165,6 +159,19 @@ export class FrameController implements AppearanceObserverDelegate, FetchRequest
 
   // Private
 
+  private async visit(url: Locatable) {
+    const location = Location.wrap(url)
+    const request = new FetchRequest(this, FetchMethod.get, location)
+
+    return new Promise<void>(resolve => {
+      this.resolveVisitPromise = () => {
+        this.resolveVisitPromise = () => {}
+        resolve()
```

Comment:
I've seen this pattern crop up across several projects. What's makes this nesting beneficial?